core/network/impl/peer_view.cpp

```diff
@@ -10,22 +10,44 @@
 #include "utils/weak_macro.hpp"
 
 namespace kagome::network {
-  inline View makeView(const LazySPtr<blockchain::BlockTree> &block_tree) {
+  static constexpr size_t MAX_VIEW_HEADS = 5;
+
+  inline std::pair<View, std::vector<primitives::BlockHash>> makeView(
+      const LazySPtr<blockchain::BlockTree> &block_tree) {
     auto last_finalized = block_tree.get()->getLastFinalized().number;
+    const auto leaves = block_tree.get()->getLeavesInfo();
 
-    std::vector<primitives::BlockHash> heads;
+    std::vector<std::pair<BlockNumber, primitives::BlockHash>> heads;
     for (const auto &bi : block_tree.get()->getLeavesInfo()) {
       if (bi.number >= last_finalized) {
-        heads.emplace_back(bi.hash);
+        heads.emplace_back(bi.number, bi.hash);
       }
     }
+    std::ranges::sort(
+        heads, [](const auto &l, const auto &r) { return l.first < r.first; });
+
+    std::vector<primitives::BlockHash> heads_;
+    heads_.reserve(std::min(MAX_VIEW_HEADS, heads.size()));
+    for (auto it = heads.rbegin(); it != heads.rend(); ++it) {
+      if (heads_.size() < MAX_VIEW_HEADS) {
+        heads_.emplace_back(it->second);
+      } else {
+        break;
+      }
+    }
+    std::ranges::sort(heads_);
```

Comment:
Do we need to sort before take and emplace?

Reply:
yep